app/main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+        self, name: str, health: int = 100, hidden: bool = False
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+    def die(self) -> None:
+        Animal.alive.remove(self)
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(target: Animal) -> None:
+        if isinstance(target, Herbivore) and target.hidden is False:
```

Comment:
You could simplify this check
```suggestion
        if isinstance(target, Herbivore) and not target.hidden:
```